wire-ios-request-strategy/Sources/Helpers/FederationDeleteManagerTests.swift

```diff
@@ -0,0 +1,196 @@
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import XCTest
+import WireDataModel
+@testable import WireRequestStrategy
+
+class FederationDeleteManagerTests: MessagingTestBase {
+    var sut: FederationDeleteManager!
+
+    override func setUp() {
+        super.setUp()
+        sut = FederationDeleteManager(syncContext: syncMOC)
+    }
+
+    override func tearDown() {
+        super.tearDown()
+        sut = nil
+    }
+
+    func testThatFederationDelete_markOneToOnConversationAsReadOnly()  throws {
+        syncMOC.performGroupedAndWait { [self] syncMOC in
+            // GIVEN
+            let domain = "other.user.domain"
+            otherUser.domain = domain
+            guard let conversation = otherUser.connection?.conversation else { return }
+
+            // WHEN
+            sut.backendStoppedFederatingWithDomain(domain: domain)
+
+            // THEN
+            XCTAssertTrue(conversation.isForcedReadOnly)
+            XCTAssertTrue(conversation.isReadOnly)
+        }
+    }
+
+
+    func testThatFederationDelete_RemovePendingConnectionRequest()  throws {
+        syncMOC.performGroupedAndWait { [self] syncMOC in
+            // GIVEN
+            let domain = "other.user.domain"
+            otherUser.domain = domain
+            otherUser.connection?.status = .pending
+
+            // WHEN
+            sut.backendStoppedFederatingWithDomain(domain: domain)
+
+            // THEN
+            XCTAssertEqual(otherUser.connection?.status, .ignored)
+        }
+    }
+
+
+    func testThatFederationDelete_RemoveMyConnectionRequest()  throws {
+        syncMOC.performGroupedAndWait { [self] syncMOC in
+            // GIVEN
+            let domain = "other.user.domain"
+            otherUser.domain = domain
+            otherUser.connection?.status = .sent
+
+            // WHEN
+            sut.backendStoppedFederatingWithDomain(domain: domain)
+
+            // THEN
+            XCTAssertEqual(otherUser.connection, nil)
+        }
+    }
+
+    func testThatFederationDelete_removeSelfUserFromSecondBackendConversation()  throws {
+        syncMOC.performGroupedAndWait { [self] syncMOC in
+            // GIVEN
+            let domain = "other.user.domain"
+            otherUser.domain = domain
+            let conversation = createGroupConversation(with: [otherUser], hostedByDomain: domain)
+
+            // WHEN
+            sut.backendStoppedFederatingWithDomain(domain: domain)
```

Comment:
it might be good to assert that the self user is a participant before WHEN.

Reply:
fixed